### ISSUE TITLE
prefer-readonly-type: Support mapped types

### ISSUE THREAD

**Issue by whatisaphone** [jonaskello/eslint-plugin-functional#106]
It would be nice if prefer-readonly-type supported mapped types.

## Repro

eslint config:

```
'functional/prefer-readonly-type': 'error'
```

typescript source code:

```
function indexed(x: { [_: string]: number }) {}
function mapped(x: { [_ in string]: number }) {}
```

### Expected behavior

The rule should flag both lines, and autofix them to:

```
function indexed(x: { readonly [_: string]: number }) {}
function mapped(x: { readonly [_ in string]: number }) {}
```

### Actual behavior

The rule flags the first line, but does not flag the second.

**Comment by RebeccaStevens** [comment]
I agree. I'll mark it as a bug that it doesn't already do this.
